[build] BoundingBox.cc: exchange is not a member of std

<!-- If you're not sure whether your problem is a bug, please ask a question at
http://answers.gazebosim.org instead.-->

## Environment
* OS Version: <!-- e.g, Ubuntu 20.04 --> Archlinux
* GCC: 12.1.0
* Source or binary build?
  <!-- If binary, which version? -->
  <!-- If source, which branch and what is the latest commit? --> Source: tag: [6.4.0](https://github.com/gazebosim/gz-rendering/tree/ignition-rendering6_6.4.0) 1481bf8
  
```
# paste log here
/build/ignition-rendering/src/gz-rendering-ignition-rendering6_6.4.0/src/BoundingBox.cc: In constructor ‘ignition::rendering::v6::BoundingBox::BoundingBox(ignition::rendering::v6::BoundingBox&&)’:
/build/ignition-rendering/src/gz-rendering-ignition-rendering6_6.4.0/src/BoundingBox.cc:48:18: error: ‘exchange’ is not a member of ‘std’
   48 |   : dataPtr(std::exchange(_texture.dataPtr, nullptr))
      |                  ^~~~~~~~
```

</details>

## Description
* Expected behavior: <!-- Tell us what you expected to happen --> No build error
* Actual behavior: <!-- What happened instead -->
build error occurs

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. clone
2. build
3. see error

## Output
<!-- Provide screenshots, console logs, backtraces, and/or anything that could
be useful to us resolving this issue -->

--- a/include/ignition/rendering/BoundingBox.hh
+++ b/include/ignition/rendering/BoundingBox.hh
@@ -18,6 +18,7 @@
 #define IGNITION_RENDERING_BOUNDINGBOX_HH_
 
 #include <memory>
+#include <utility>
 
 #include <ignition/common/SuppressWarning.hh>
 

